robot.py

```diff
@@ -44,6 +50,12 @@ def input_gate(self, name, data):
         dt = self.logger.write(self.stream_id, bytes(str((name, data)),'ascii'))  # TODO single or mutiple streams?
         self.queue.put((dt, name, data))
 
+    def execute(self, msg_id, data):
+        assert self.stream_id_out is not None
+        dt = self.logger.write(self.stream_id_out, bytes(str((msg_id, data)),'ascii'))
```

Comment:
Why not this instead?
```python
dt = self.logger.write(self.stream_id_out, bytes(msg_id, data))
```
Well, now that I look at it closer - we are logging the data as text? Is there a specific reason for that?

Reply:
And we are storing different messages into the same stream? I think we should give this some more thought. I was somehow expecting that a single stream inside the log file would contain only one type of message.

Reply:
Well, the text is temporary - just to get it moving. Then I would like to return to numpy arrays/bytes.

One stream - well, that was maybe misunderstanding from #42 where you wrote "I'd go for one stream and made a log of this stream. For me, this would be THE LOG."
???

Reply:
> Well, the text is temporary - just to get it moving. Then I would like to return to numpy arrays/bytes.

Ok.

> One stream - well, that was maybe misunderstanding from #42 where you wrote "I'd go for one stream and made a log of this stream. For me, this would be THE LOG."
> ???

That is misunderstanding regarding the term "stream". There we were talking about communication between robot and application. There I was proposing just one, let's say, _channel/pipe/stream_. In this pipe there would be several _topics_. Each message would have `(topic, timestamp, data)` structure, which would be saved to THE LOG. At the beginning of the log there would be a description of the format of the `data` section belonging to each topic (different topic can have same data type).

If it does not make sense, let's discuss this in person.